src/fast-jsonapi-serializer/relationship_config.cr

```diff
@@ -0,0 +1,57 @@
+module FastJSONAPISerializer
+  class RelationshipConfig
+    getter name : Symbol?
+    getter children : Array(RelationshipConfig)?
+    getter embed = true
+
+    def initialize(@name = nil, @children = nil)
+    end
+
+    def embed(@embed)
+      children.try &.each { |child| child.embed(@embed) }
+
+      self
+    end
+
+    def have_relation?(name : Symbol)
+      children.try &.any? { |rel| rel.name == name }
+    end
+
+    def nested(name : Symbol)
+      null_rel = self.class.new()
+
+      children.try &.find(null_rel) { |rel| rel.name == name } || null_rel
+    end
+
+    def empty?
+      children.nil? || children.try &.empty?
+    end
+
+    def traverse(path)
+      children.not_nil!.find { |child| child.name == path }.not_nil!
```

Comment:
Should we throw an error here with a message to inform the user if they don't have child nodes?